pkg/qe-tests/cypress/integration/models/providerVmware.ts

```diff
@@ -153,16 +153,18 @@ export class ProviderVmware extends Provider {
     this.populate(providerData);
   }
 
-  edit(providerData: VmwareProviderData): void {
-    const { name, hostname, username, password } = providerData;
+  edit(providerData: VmwareProviderData, editProviderData: VmwareProviderData): void {
+    const { name } = providerData;
+    const { hostname, username, password } = editProviderData;
     ProviderVmware.openList();
     applyAction(name, editButton);
     this.fillHostname(hostname);
```

Comment:
Will the user always send updated values for all these fields ? Hostname, username ?
What if I just want to update hostname and not rest of the fields ?

Reply:
Thanks a lot @sshveta for the Review !! 
The test case test_provider_update requirement was to edit all the three fields i.e Hostname, username and the password.
So, we have designed the test case accordingly.
And yes,  code can be modified if we just want to update any of the required fields. It's just a matter of using if-else or any other way out. But, as it wasn't a mandatory condition so, we implemented this way.